Library/Homebrew/extend/os/mac/hardware/cpu.rb

```diff
@@ -13,7 +13,7 @@ class << self
       def type
         case sysctl_int("hw.cputype")
         when MachO::Headers::CPU_TYPE_I386
-          :intel
+          in_rosetta2? ? :arm : :intel
```

Comment:
This is going to break scenarios where people do want to use Intel Homebrew on an arm64 machine (e.g. `arch -x86_64 brew`).